crates/rpc/rpc/src/eth/error.rs

```diff
@@ -154,7 +154,19 @@ pub enum InvalidTransactionError {
     SenderNoEOA,
     /// Thrown during estimate if caller has insufficient funds to cover the tx.
     #[error("Out of gas: gas required exceeds allowance: {0:?}")]
-    OutOfGas(U256),
+    BasicOutOfGas(U256),
+    /// TODO?
+    //#[error("Out of gas:")]
+    //MemoryLimitOutOfGas(U256),
+    /// As BasicOutOfGas but thrown when gas exhausts during memory expansion.
+    #[error("Out of gas: gas exhausts during memory expansion, gas required exceeds allowance: {0:?}")]
+    MemoryOutOfGas(U256),
+    /// As BasicOutOfGas but thrown when gas exhausts during precompiled contract execution.
+    #[error("Out of gas: gas exhausts during precompiled contract execution, gas required exceeds allowance: {0:?}")]
+    PrecompileOutOfGas(U256),
+    /// TODO?
+    // #[error("Out of gas: {0:?}")]
+    // InvalidOperandOutOfGas(U256),
```

Comment:
I don't understand why this error is of type OOG in revm. Should be handled? And which error message I should write?